### PR TITLE
Feature/iteration2

### PR DESCRIPTION
**Pull request by knees4bees**
## What does this PR do? ##
- Gets close to completing iteration 2 (still need to attach event listener to rooms container and POST bookings data, need to add a couple media queries, and need to fix weird accessibility issue).
- Adds `RoomsRepo.js` and associated test; also moves a few functions to the `helpers.js` file.
- Adds display of available rooms for a selected date with ability to filter based on room types.
- Makes styling updates for improved accessibility and responsiveness.

## How should it be tested? ##
1. Fetch and install my code:
  - `cd` into the directory for my project
  - `git fetch`
  - `git checkout feature/iteration2`
  - in a second Terminal tab, `npm start` (you may need to `ctrl c` to stop your own local server from running your version of the project)
2. Confirm that all tests are passing when you run `npm test`.
3. In a browser, go to `http://localhost:8080/` and confirm that the needed data is being displayed. Try searching for rooms on different days, and then try filtering by different room types. Confirm that if no room types are selected an apology message is displayed (it may not be fierce enough though).
4. Take a look at the class file `RoomsRepo.js` and see if you spot anything that looks like a bad move or a code smell.

I'm getting a weird error in my Lighthouse accessibility audit where it says "form elements do not have associated labels". I added labels to both form elements on the landing page, and I'm still getting this error. I checked that each element has an `id` that matches its associated label's `for` attribute. Thoughts?

### CODE REVIEW

File: src/css/rooms.scss [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  margin: 0;
+  display: flex;
+  justify-content: space-between;
+  padding: 5px;
+  min-height: 150px;
+  min-width: 300px;
+  background-color: white;
+  border-radius: 5px;
+  box-shadow: 0px 0px 7px $main-medium;
+
```
Comment (jgiwinski) [inline src/css/rooms.scss:59]: if you need more variables to check off on the specs then you might be able to make your box-shadow a var or even a mixin. just make the parameter a different color if you want the shadow to be dynamic. (i completely forgot about making mixins dynamic and parameters until trravis reminded me about them in my eval)

File: src/css/rooms.scss [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  // margin-top: 15px;
+  margin: 15px;
+  padding: 10px;
+  background-color: $accent-medium;
+  color: white;
+  font-size: 1.15rem;
+  font-weight: bold;
+  border: 0.1rem solid $accent-dark;
+  border-radius: 5px;
+  box-shadow: 0px 0px 7px $accent-dark;
```
Comment (jgiwinski) [inline src/css/rooms.scss:92]: yea here you could pass in the $accent-dark into a box-shadow(color) mixin.

File: src/helpers.js [hunk 3]
```diff
@@ -23,4 +27,23 @@ function createBookings(rawBookingsData, rawCustomersData, rawRoomsData) {
...
+    rooms.push(room);
+  });
+
+  return rooms;
+}
+
+function formatDate(date) {
+  const prettyDate = date.replace(/-/g, '/');
+  return prettyDate;
+}
```
Comment (jgiwinski) [inline src/helpers.js:46]: ahh the cursed replace/replaceAll function

File: src/scripts.js [hunk 4]
```diff
@@ -96,7 +129,109 @@ function renderBookings(bookingsRepo) {
...
+  }
+
+  let bidet = '';
+
+  roomsRepo.rooms.forEach(room => {
+    if (room.bidet) {
+      bidet = '<i class="fas fa-toilet"></i> bidet';
+    } else {
+      bidet = '';
+    }
```
Comment (jgiwinski) [inline src/scripts.js:161]: i love all your icons and visuals. gives UX/UI some extra points in my book

File: src/scripts.js [hunk 5]
```diff
@@ -96,7 +129,109 @@ function renderBookings(bookingsRepo) {
...
+  return types;
+}
+
+function getRooms(types) {
+  let rooms = [];
+
+  types.forEach(type => {
+    const roomsOfThisType = allRooms.filterByType(type).rooms;
+    rooms = [...rooms, ...roomsOfThisType];
+  });
```
Comment (jgiwinski) [inline src/scripts.js:205]: Nice use of spread. Feel like it really cleans this function up

File: test/booking-test.js [hunk 6]
```diff
@@ -4,9 +4,25 @@ const expect = chai.expect;
...
+  room2 as testRoom2, 
+  room3 as testRoom3 
+} from './test-data';
+
+import { 
+  booking1 as testBooking1, 
+  booking2 as testBooking2, 
+  booking3 as testBooking3, 
+  booking4 as testBooking4
+} from './test-data';
```
Comment (jgiwinski) [inline test/booking-test.js:26]: A clean update. I likes. ^^

**Review by jgiwinski — APPROVED** [review]
Looks like all the pieces are coming together!! You'll be able to knock it all out today for sure. All your class tests are really solid and clearly structured. Nothing to worry about there. They all pass on my end, too! 

Put a small note on maybe adding a mixin to your files for the boxshadow. it might not NEED it, but it would be some good practice to use.

File: src/css/rooms.scss (hunk 1 above)
Comment (knees4bees) [reply src/css/rooms.scss:59]: Great idea, thanks!!

File: src/helpers.js (hunk 3 above)
Comment (knees4bees) [reply src/helpers.js:46]: RIP `replaceAll`

**Comment by knees4bees** [comment]
Thanks, @jgiwinski ! These are great comments. Thanks for taking the time to poke around and make suggestions. I appreciate you!

**Pull request merged**